Makefile

```diff
@@ -10,8 +10,17 @@ SHIM ?= "${GOPATH}/src/github.com/clearcontainers/shim/cc-shim"
 # The time limit in seconds for each test
 TIMEOUT ?= 5
 
-functional:
-	ginkgo ./functional/ -- -runtime "${RUNTIME}" -proxy "${PROXY}" -shim "${SHIM}" -timeout ${TIMEOUT}
+TESTS_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+
+GINKGO_PATH = ${GOPATH}/bin/ginkgo
+
+ginkgo:
+ifeq ($(wildcard $(GINKGO_PATH)),)
```

Comment:
I don't find this readable nor obvious, so can't give my lgtm, but I'm sure others can.